### DISCUSSION TITLE
nnbd migration

### DISCUSSION THREAD

**Discussion by ali-thowfeek** [ali-thowfeek/simple_html_css_flutter#10] [General]
@nohli I thought of maintaining a separate branch for [nnbd pre release](https://github.com/ali-thowfeek/simple_html_css_flutter/tree/pre-release-nnbd). What do you think? Shall we merge #9 into that?

**Comment by nohli** [comment]
@ali-thowfeek yes, that would be good!

**Reply by ali-thowfeek** [reply]
Published the [nnbd pre-release](https://pub.dev/packages/simple_html_css/versions/3.0.0-nullsafety.0). Thank you very much ♥️

**Comment by ali-thowfeek** [comment]
Hey @nohli, there seems to be some [problems](https://pub.dev/packages/simple_html_css/versions/3.0.0-nullsafety.0/score) which I can't quite get my head around.

**Reply by nohli** [reply]
Ok will check

**Reply by nohli** [reply]
@ali-thowfeek which version did you use for uploading? Could you try `flutter upgrade` please?

Background: in the dartdoc log it says `flutter: 1.25.0-8.1.pre `, but the latest beta version is `1.25.0-8.3.pre`.
Idk if this is the version on your machine.

The log says that `2.12.0-133.2.beta` is not compatible with `dart: ">=2.12.0-133.7.beta <3.0.0"` (which you would get from pub get with the latest flutter beta, and which would probably work - but lock files don't get uploaded to git). Maybe before uploading the package, an old flutter version did pub get and fetched old constraints.

**Reply by ali-thowfeek** [reply]
Well, after i switched to the `beta` channel i did run `flutter upgrade`.  Currently the dart --version is `Dart SDK version: 2.12.0-133.7.beta` and `Flutter (Channel beta, 1.25.0-8.3.pre`

**Reply by nohli** [reply]
That sounds good.
And you've uploaded with this version?
This was my first guess...then I try to check further.

**Reply by ali-thowfeek** [reply]
Hey, did you happen to check this [log](https://pub.dev/documentation/simple_html_css/3.0.0-nullsafety.0/log.txt). In the end it says 
```
Because simple_html_css depends on html_unescape >=2.0.0-nullsafety.0 which requires SDK version >=2.12.0-133.7.beta <3.0.0, version solving failed.
```

[html_unescape's score](https://pub.dev/packages/html_unescape/versions/2.0.0-nullsafety.0/score) also seems to have kinda the same [problem](https://pub.dev/documentation/html_unescape/2.0.0-nullsafety.0/log.txt)

Do we have to downgrade to fix this?

**Reply by nohli** [reply]
If we downgrade, we would import legacy libraries (not null safe). I will open a pull request in that package as well.
It's interesting that pub.dev seems to have an older flutter version.
Edit: Hope, this helps: https://github.com/filiph/html_unescape/pull/10

**Reply by nohli** [reply]
https://github.com/dart-lang/pub-dev/issues/4452#issuecomment-768877383

Even if we do nothing, eventually the score will be fine again.

**Reply by ali-thowfeek** [reply]
Thank you so much for all the efforts you've put in. Really appreciate it.

**Reply by nohli** [reply]
We'll use your package in a production app, so thanks to you ;)

**Reply by nohli** [reply]
FYI they've updated pub.dev to flutter `1.25.0-8.3.pre` and fixed the score ;)
